File: Supplier wrapper BK/NI-RFSG19.1DotNETWrapper/NI-RFSG19.1DotNETWrapper/19.1/Examples/SingleToneGeneration/vb/MainForm.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports NationalInstruments.ModularInstruments.Interop


Namespace NationalInstruments.Examples.SingleToneGeneration
    Partial Public Class MainForm
        Inherits Form
        Private _rfsgSession As niRFSG

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub startGeneration()
            Dim resourceName As String
            Dim frequency As Double, power As Double
            resourceName = ResourceNameTextBox.Text
            frequency = CDbl(frequencyNumeric.Value)
            power = CDbl(powerLevelNumeric.Value)

            errorTextBox.Text = "No error."
            ' Initialize the NIRfsg session
            If _rfsgSession Is Nothing Then
                _rfsgSession = New niRFSG(resourceName, True, False)
            End If
            _rfsgSession.ConfigureRF(frequency, power)
            _rfsgSession.Initiate()
            rfsgStatusTimer.Start()
        End Sub

        Private Sub closeSession()
            If _rfsgSession IsNot Nothing Then
                _rfsgSession.close()
            End If
        End Sub
        Private Sub stopGeneration()
            Try
                If _rfsgSession IsNot Nothing Then
                    ' Disable the output.  This sets the noise floor as low as possible.
                    _rfsgSession.ConfigureOutputEnabled(False)

                    ' Close the RFSG NIRfsg session
                    _rfsgSession.close()

                    startButton.Enabled = True
                    updateButton.Enabled = False
                    stopButton.Enabled = False
                    rfsgStatusTimer.Enabled = False
                End If
                _rfsgSession = Nothing
            Catch ex As Exception
                errorTextBox.Text = "Error in StopGeneration(): " & ex.Message
            End Try
        End Sub
        Private Sub CheckGeneration()
            ' Check the status of the RFSG 
            Dim isDone As Boolean = False
            _rfsgSession.CheckGenerationStatus(isDone)
        End Sub
        Private Sub rfsgStatusTimer_Tick(ByVal sender As Object, ByVal e As System.EventArgs)
            Try
                CheckGeneration()
            Catch ex As System.Exception
                If ex.Message <> "" Then
                    errorTextBox.Text = ex.Message
                Else
                    errorTextBox.Text = "Error"
                End If
            End Try
        End Sub

        Private Sub startButton_Click(ByVal sender As Object, ByVal e As EventArgs)
            Try
                errorTextBox.Text = "No Error"
                startButton.Enabled = False
                updateButton.Enabled = True
                stopButton.Enabled = True
                rfsgStatusTimer.Enabled = True
                startGeneration()
            Catch ex As System.Exception
                If ex.Message <> "" Then
                    errorTextBox.Text = ex.Message
                Else
                    errorTextBox.Text = "Error"
                End If
            Finally
                If _rfsgSession IsNot Nothing Then
                    _rfsgSession.Abort()
                    _rfsgSession.reset()
                End If
            End Try
        End Sub

        Private Sub stopButton_Click(ByVal sender As Object, ByVal e As EventArgs)
            stopGeneration()
        End Sub
        Private Sub MainForm_FormClosing(ByVal sender As Object, ByVal e As System.Windows.Forms.FormClosingEventArgs)
            stopGeneration()
        End Sub

        Private Sub updateButton_Click(ByVal sender As Object, ByVal e As EventArgs)
            UpdateGeneration()
        End Sub

        Private Sub UpdateGeneration()
            Dim frequency As Double, power As Double
            Try
                startButton.Enabled = True
                ' Read in all the control values 
                frequency = CDbl(frequencyNumeric.Value)
                power = CDbl(powerLevelNumeric.Value)

                ' Abort generation 
                _rfsgSession.Abort()

                ' Configure the instrument 
                _rfsgSession.ConfigureRF(frequency, power)

                ' Initiate Generation 
                _rfsgSession.Initiate()

                ' Start the status checking timer 

                startButton.Enabled = False
            Catch ex As Exception
                If ex.Message <> "" Then
                    errorTextBox.Text = ex.Message
                Else
                    errorTextBox.Text = "Error"
                End If
            End Try
        End Sub

    End Class
End Namespace

File: Supplier wrapper BK/NI-RFSG19.1DotNETWrapper/NI-RFSG19.1DotNETWrapper/19.1/Examples/SingleToneGeneration/cs/MainForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using NationalInstruments.ModularInstruments.Interop;


namespace NationalInstruments.Examples.SingleToneGeneration
{
    public partial class MainForm : Form
    {
        niRFSG _rfsgSession;
        /// <summary>
        /// 
        /// </summary>
        public MainForm()
        {
            InitializeComponent();
        }

        void startGeneration()
        {
            string resourceName;
            double frequency, power;     
            resourceName = ResourceNameTextBox.Text;
            frequency = (double)frequencyNumeric.Value;
            power = (double)powerLevelNumeric.Value;

            errorTextBox.Text = "No error.";
            // Initialize the NIRfsg session
            if (_rfsgSession == null)
            {
                _rfsgSession = new niRFSG(resourceName, true, false);
            }
             _rfsgSession.ConfigureRF(frequency, power);
             _rfsgSession.Initiate();
             rfsgStatusTimer.Start();
        }

        void closeSession()
        {
            if (_rfsgSession != null)
            {
                _rfsgSession.close();
            }
        }
        void stopGeneration()
        {
            try
            {
                if (_rfsgSession != null)
                {
                    // Disable the output.  This sets the noise floor as low as possible.
                    _rfsgSession.ConfigureOutputEnabled(false);

                    // Close the RFSG NIRfsg session
                    _rfsgSession.close();

                    startButton.Enabled = true;
                    updateButton.Enabled = false;
                    stopButton.Enabled = false;
                    rfsgStatusTimer.Enabled = false;
                }
                _rfsgSession = null;
            }
            catch (Exception ex)
            {
                errorTextBox.Text = "Error in StopGeneration(): " + ex.Message;
            }
        }
        void CheckGeneration()
        {
            // Check the status of the RFSG 
            bool isDone = false;
            _rfsgSession.CheckGenerationStatus(out isDone);
        }
        private void rfsgStatusTimer_Tick(object sender, System.EventArgs e)
        {
            try
            {
                CheckGeneration();
            }
            catch (System.Exception ex)
            {
                if (ex.Message != "")
                {
                    errorTextBox.Text = ex.Message;
                }
                else
                {
                    errorTextBox.Text = "Error";
                }
            }
        }
        
        private void startButton_Click(object sender, EventArgs e)
        {
            try
            {                
                errorTextBox.Text = "No Error";
                startButton.Enabled = false;
                updateButton.Enabled = true;
                stopButton.Enabled = true;
                rfsgStatusTimer.Enabled = true;
                startGeneration();               
            }
            catch (System.Exception ex)
            {
                if (ex.Message != "")
                {
                    errorTextBox.Text = ex.Message;
                }
                else
                {
                    errorTextBox.Text = "Error";
                }
            }
            finally
            {
                if (_rfsgSession != null)
                {
                    _rfsgSession.Abort();
                    _rfsgSession.reset();
                }
            }
        }

        private void stopButton_Click(object sender, EventArgs e)
        {
            stopGeneration();
        }
        private void MainForm_FormClosing(object sender, System.Windows.Forms.FormClosingEventArgs e)
        {
            stopGeneration();
        }

        private void updateButton_Click(object sender, EventArgs e)
        {
            UpdateGeneration();
        }

        private void UpdateGeneration()
        {
            double frequency, power;
            try
            {
                startButton.Enabled = true;
                // Read in all the control values 
                frequency = (double)frequencyNumeric.Value;
                power = (double)powerLevelNumeric.Value;

                // Abort generation 
                _rfsgSession.Abort();

                // Configure the instrument 
                _rfsgSession.ConfigureRF(frequency, power);

                // Initiate Generation 
                _rfsgSession.Initiate();

                // Start the status checking timer 
                startButton.Enabled = false;

            }
            catch (Exception ex)
            {
                if (ex.Message != "")
                {
                    errorTextBox.Text = ex.Message;
                }
                else
                {
                    errorTextBox.Text = "Error";
                }
            }
        }

    }
}
